services/apps/portal/src/application/application-document/application-document.controller.spec.ts

```diff
@@ -0,0 +1,150 @@
+import { classes } from '@automapper/classes';
+import { AutomapperModule } from '@automapper/nestjs';
+import { createMock, DeepMocked } from '@golevelup/nestjs-testing';
+import { BadRequestException } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import { ClsService } from 'nestjs-cls';
+import { mockKeyCloakProviders } from '../../../test/mocks/mockTypes';
+import { ApplicationProfile } from '../../common/automapper/application.automapper.profile';
+import { User } from '../../user/user.entity';
+import { ApplicationDocumentController } from './application-document.controller';
+import { ApplicationDocument } from './application-document.entity';
+import { ApplicationDocumentService } from './application-document.service';
+
+describe('ApplicationDocumentController', () => {
+  let controller: ApplicationDocumentController;
+  let appDocumentService: DeepMocked<ApplicationDocumentService>;
+
+  const mockDocument = new ApplicationDocument({
+    uploadedBy: new User(),
+  });
+
+  beforeEach(async () => {
+    appDocumentService = createMock<ApplicationDocumentService>();
+
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [
+        AutomapperModule.forRoot({
+          strategyInitializer: classes(),
+        }),
+      ],
+      controllers: [ApplicationDocumentController],
+      providers: [
+        ApplicationProfile,
+        {
+          provide: ApplicationDocumentService,
+          useValue: appDocumentService,
+        },
+        {
+          provide: ClsService,
+          useValue: {},
+        },
+        ...mockKeyCloakProviders,
+      ],
+    }).compile();
+    controller = module.get<ApplicationDocumentController>(
+      ApplicationDocumentController,
+    );
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+  });
+
+  it('should return the attached document', async () => {
+    const mockFile = {};
+    const mockUser = {};
+
+    appDocumentService.attachDocument.mockResolvedValue(mockDocument);
+
+    const res = await controller.attachDocument('file', 'certificateOfTitle', {
+      isMultipart: () => true,
+      file: () => mockFile,
+      user: {
+        entity: mockUser,
+      },
+    });
+
+    //expect(res.mimeType).toEqual(mockDocument.document.mimeType);
```

Comment:
do we need this?

Reply:
I figure it would be re-enabled when we actually hook up document uploading, but I dont know